sql/sql_acl.cc

```diff
@@ -14312,6 +14313,50 @@ bool acl_authenticate(THD *thd, uint com_change_user_pkt_len)
 
   if (res == CR_OK_HANDSHAKE_COMPLETE)
     thd->get_stmt_da()->disable_status();
+  else
+  if (redirect_enabled)
+  {
+    int total_size = 18
+                   + strlen(redirect_server_host)
+                   + 1 + strlen(redirect_server_port)
+                   + 6 + strlen(sctx->user)
+                   + (!strcmp(redirect_server_ttl, "0") ? 0 : (5 + strlen(redirect_server_ttl)));;
+
+    if (total_size >= MAX_REDIRECTION_LEN)
+    {
+      sql_print_error("error", ("redirection info is too large to return to client (len= %lu)",
+                     total_size));
+      my_ok(thd);
+    }
+    else
+    {
+      char *msg = new char[MAX_REDIRECTION_LEN];
```

Comment:
could just do a `char  msg[MAX_REDIRECTION_LEN];` and what the snprintf response for the reaching of the end of the buffer.

Reply:
Nice catch for the statically allocated msg; but I would like to keep the check of reaching message end, as the OK packet is needed to inform client it is authenticated even if redirection message cannot be provided from the server side.

Reply:
really didn't want to allocate a buffer(which is not to be used) if redirection info exceeds maximum buffer length... any reason we prefer the snprintf response over the total_size >= MAX_REDIRECTION_LEN check?

Reply:
readability mainly. Its easier to verify that its all fitting in the buffer than counting all the bits. If the URL is extended with other aspects it will be easier to maintain that verification.

There are hooks you could do on the global variables to ensure its less than this length when they are changes however given a reasonable length I think its a bit overboard to do so.

Reply:
Changes made as per comments stated.